reconcile/openshift_namespaces.py

```diff
@@ -1,127 +1,136 @@
 import logging
 
-import reconcile.utils.gql as gql
+from typing import List, Dict, Optional, Tuple, Any, Iterable, Mapping
 import reconcile.utils.threaded as threaded
-import reconcile.openshift_base as ob
 import reconcile.queries as queries
 
-from reconcile.utils.openshift_resource import ResourceInventory
 from reconcile.utils.oc import OC_Map
 from reconcile.utils.oc import StatusCodeError
 from reconcile.utils.defer import defer
 from reconcile.utils.sharding import is_in_shard
 
 
-QUERY = """
-{
-  namespaces: namespaces_v1 {
-    name
-    cluster {
-      name
-      serverUrl
-      jumpHost {
-          hostname
-          knownHosts
-          user
-          port
-          identity {
-              path
-              field
-              format
-          }
-      }
-      automationToken {
-        path
-        field
-        format
-      }
-      internal
-      disable {
-        integrations
-      }
-    }
-  }
-}
-"""
-
 QONTRACT_INTEGRATION = 'openshift-namespaces'
 
+NS_STATUS_PRESENT = "present"
+NS_STATUS_ABSENT = "absent"
+
+NS_ACTION_DELETE = "delete"
+NS_ACTION_CREATE = "create"
+
+
+def get_desired_state(
+        namespaces: Iterable[Dict[str, Any]]) -> List[Dict[str, str]]:
```

Comment:
Can this be `Iterable[Mapping[str, Any]]` or was there a specific need to leave this one?

Reply:
Sorry. I missed this one. Fixed.